zebra/zebra_rnh.c

```diff
@@ -608,7 +608,8 @@ zebra_rnh_resolve_nexthop_entry(struct zebra_vrf *zvrf, afi_t afi,
 				continue;
 			}
 
-			if (CHECK_FLAG(re->status, ROUTE_ENTRY_QUEUED)) {
+			if (CHECK_FLAG(re->status, ROUTE_ENTRY_QUEUED) &&
```

Comment:
ok I don't understand how this fixes the `problem`.  If I am running nexthop tracking on a address at this point.  We have probably just gotten the callback for a route installed that going to possibly match this address.  If we don't match it at this point we should match it after nexthop tracking gets invoked again after the Queued entry is finally installed.  What is the actual code path that is going wrong?  Is the route staying queued?  If so then that is the actual problem that needs to be fixed.

Reply:
In the case we ran into, the ROUTE_ENTRY_QUEUED flag stayed and did not get cleared at all (even after several hours). There is certainly a bug with the flag.  The issue occurred in an automated, scale testing multiple times.
  
The route was installed in FIB. As the flag ROUTE_ENTRY_INSTALLED is much more important, the idea is to make the route eligible for nexthop resolution as long as that flag is set. We tried the patch, and did not see the issue afterwards.

Reply:
Then the proper fix is finding and fixing the unsetting of the Queued flag, not this.  Having said that I have fixed several places where the Queued flag wasn't cleared over the last couple of years are you using an older version of frr?

Reply:
The version is 8.4.4. It's good to know there have been fixes in that area.  The patch is a workaround, but is beneficial so that a modest issue does not result in severe network disruption.

Reply:
I recommend that you fix the actual problem instead of this.

Reply:
Shouldn't we iterate over the code and make sure we do:
```c
UNSET_FLAG(re->status, ROUTE_ENTRY_QUEUED);
```
After/before we do:
```c
SET_FLAG(re->status, ROUTE_ENTRY_INSTALLED);
```

Reply:
that's my point, I believe that 8.4.X doesn't have all the route entry queued fixes put in place.  We are missing a unset of the queued flag somewhere.  In the event that master does not have the right flag unsetting then we should fix that.  But I would need more data to go further here